### DISCUSSION TITLE
Filer leveldb3 performance considerations

### DISCUSSION THREAD

**Discussion by divanikus** [seaweedfs/seaweedfs#2206] [General]
We seem to hit some limitations of `filer`'s `leveldb3` storage and need some help on how to overcome them.
Our system is writing data to daily buckets. Once we have found that `chunk` and `put` operations suddenly doubled their execution time.

![image](https://user-images.githubusercontent.com/3438036/126405591-551a4128-7916-43ac-be77-241ccb1fb5af.png)

As you may see on the graph, at first it was at 50 ms, than doubled to 100 ms, later is became 200 ms and suddenly 400 ms. You may say that's 99th percentile and not a big deal, but we see all our ops are slowed down by exactly that time! Eventhough 80th percentile doesn't look that bad, slowdown is visible there too.

![image](https://user-images.githubusercontent.com/3438036/126405890-ad3831ba-0df6-4bf2-8ac4-6ca18a30db12.png)

Seems like it is related to number of files in the bucket, because starting a fresh bucket resets the request times to normal values. 

I guess we are doing something wrong, like writing too many files into one folder or something like that. Can you please give us a hint on what we can do to reduce filer requests time?

**Comment by chrislusf** [comment]
does separating into more buckets help?

**Comment by divanikus** [comment]
Currently we have one bucket per day. We thought on splitting it to lesser intervals like 6 hours per bucket or something like that. Simply starting a fresh bucket with zero files resets the request time.

**Comment by chrislusf** [comment]
How many files in a folder?

**Reply by divanikus** [reply]
15-20 millions. Would it be better to write them into subfolders?

**Reply by chrislusf** [reply]
yes.

**Reply by divanikus** [reply]
It didn't helped unfortunatelly. Still getting request times up to 1 second. We have splitted bucket to one thousand subfolders with I guess equal file distribution between them.

**Reply by chrislusf** [reply]
how about trying rockdb version?

**Reply by divanikus** [reply]
According to docs, seems like it doesn't support fast bucket deletion? Wouldn't it suffer on such large collection?

**Reply by chrislusf** [reply]
adding fast bucket deletion should be relatively easy.
